providers/http/server/src/models/endpoint.go

```diff
@@ -16,9 +16,15 @@ type Resource struct {
 }
 
 type ResourceProperties struct {
-	URL string `json:"URL"`
+	URL    string `json:"URL" binding:"required"`
+	Method string `json:"method"`
 }
 
 type Endpoint struct {
 	URL string `json:"title"`
 }
+
+type SaveResponse struct {
+	Status     string
+	StatusCode int
+}
```

Comment:
For the return contract, the SaveResponse should contain all of the `resource` properties of the request, as well as the 'read-only' properties `status` & `statusCode` - e.g.:
```ts
{
  resource: {
    type: 'request@v1'
    properties: {
      url: 'https://www.microsoft.com'
      status: '200 OK'
      statusCode: 200
    }
  }
}
```

Reply:
Updated:
![image](https://user-images.githubusercontent.com/5220939/192009436-f862876f-df0d-40ca-9931-2e83b23dacd1.png)